.github/workflows/tests.yml

```diff
@@ -0,0 +1,52 @@
+name: Tests
+
+on:
+  push:
+    branches: [ master ]
+    paths:
+    - '**.cpp'
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+    paths:
+    - '**.cpp'
+
+jobs:
+  PSI-cpp-tests:
+
+    runs-on: ubuntu-latest
```

Comment:
Maybe we should the tests on MacOS(Windows?) too.

Reply:
I'm not sure the C++ code even compiles on Windows. It should be possible to get it to compile, but I don't think it's a priority for now.

Reply:
done, added ubuntu and macos